### ISSUE TITLE
BUG: Multiple aliases on the same column not behaving as expected

### ISSUE THREAD

**Issue by zbrookle** [ibis-project/ibis#2229]
``` python
column = table.some_column
table.projection(
        [
            column.name("alias1"),
            column.name("alias2"),
            column.name("alias3"),
        ]
    )
```

I think the expected behavior would be a table expression with:
```
Selection[table]
  table:
    Table: ref_0
  selections:
    alias1 = Column[float64*] 'some_column' from table
      ref_0
    alias2 = Column[float64*] 'some_column' from table
      ref_0
    alias3 = Column[float64*] 'some_column' from table
      ref_0
```

However, the result I'm getting is:
```
Selection[table]
  table:
    Table: ref_0
  selections:
    alias1 = Column[float64*] 'some_column' from table
      ref_0
    alias1 = Column[float64*] 'some_column' from table
      ref_0
    alias1 = Column[float64*] 'some_column' from table
      ref_0
```

**Comment by jreback** [comment]
reverting the PR

**Comment by gforsyth** [comment]
Closed via #2284
